changelog/pragma-mangle-aggregate.dd

```diff
@@ -0,0 +1,43 @@
+`pragma(mangle,...)` can now be used on aggregates to control the mangling of symbols that use that type.
+
+The syntax is `pragma(mangle, str [,templ] ) declaration;` where `str` is an expression that evaluates to a string
+and `templ` is an optional parameter that evaluates to a symbol and `declaration` is the aggregate whose mangling
+is to be altered. If `templ` is a template instance, `declaration` is treated as being a template with the template
+signature and parameters of `templ`.
+
+To bind C++'s `std::function`
+---
+template RawFunctionOf(F)
+{
+    F f;
+    alias RawFunctionOf = typeof(*f);
+}
+
+extern(C++, "std")
+{
+    extern(C++, struct)
+    class std_function(F)
+    {
+
+    }
+}
+template ScopeClass(C , string name)
+{
+    enum ns = __traits(getCppNamespaces,C);
+    extern(C++, class)
+    {
+        extern(C++,(ns))
+        {
+            pragma(mangle, name, C)
+            struct ScopeClass
+            {
+                char[__traits(classInstanceSize, C)] buffer;
+                //... all the things ...
+            }
+        }
+    }
+}
+
+// Mangling corresponds to `void funk(std::function<void(int)> a);
+extern(C++) void funk( ScopeClass!(std_function!(RawFunctionOf!(void function(int))),"function") a );
```

Comment:
Half of the line length seems to be to get a function type (which was possible in D1 but not in D2).
Perhaps just make the function its own 1-line function, and use `typeof` ?